Amplify/Categories/DataStore/Model/Temporal/TemporalFormat.swift

```diff
@@ -7,37 +7,92 @@
 
 import Foundation
 
-/// `Enum` that represent the different options for formatting ISO8601 values.
-/// Each implementation of a `TemporalSpec` is responsible for converting the enum value
-/// to a corresponding format string.
-public enum TemporalFormat: CaseIterable {
+public struct TemporalFormat {
+    let dateFormat: String
+    let dateTimeFormat: String
+    let timeFormat: String
 
-    case short
-    case medium
-    case long
-    case full
```

Comment:
does changing the enum case to the static properties impact customers? are we thinking along the lines that TemporalFormat wasn't meant to be used publicly?

Reply:
Yes, but only in one specific case that is very improbable because there would be no reason for a customer to do so. It would only break if they're exhaustively switching over `TemporalFormat` without a `default` case. 
```swift
switch temporalFormat {
case .short: ...
case .medium: ...
case .long: ...
case .full: ...
}
```

If that occurred, they would have to add a `default` case to satisfy exhaustiveness.

However, we will almost certainly have to add new cases in the future. This protects us against those changes being breaking changes. The description in the original PR comment above discusses the rationale behind this change.

> TemporalFormat in the previous implementation was an enum. This has been changed to a struct with static members. This change does have the possibility to break client code, although the likelihood is exceedingly low. The only situation this could happen is if client code is switching over all cases exhaustively; the change to a struct means that this check would no longer be exhaustive. Switching over the formats as they exist today would serve no purpose. Even if this is happening somewhere, the fix is adding a default case to the end of the switch. The main reason behind this change is to support the future expansion of further date formats as documented in https://github.com/aws-amplify/amplify-ios/issues/1583 and https://github.com/aws-amplify/amplify-ios/discussions/1713, without breaking client code.

Reply:
thanks for explaining, extendability sounds like it will help a lot with platform parity across Android/Flutter/JS since right now most platforms i believe do not change the user's input, while iOS normalizes to these 4 formats. A potential solution for platform parity is to find the lowest common denominator and if iOS can increase that denominator then we'll have a wider range of coverage